openedx/core/lib/api/tests/test_authentication.py

```diff
@@ -193,24 +199,48 @@ def test_post_form_token_removed_failing_auth(self):
         response = self.csrf_client.post('/oauth2-test/', HTTP_AUTHORIZATION=auth)
-        self.assertIn(response.status_code, (status.HTTP_401_UNAUTHORIZED, status.HTTP_403_FORBIDDEN))
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
 
     @unittest.skipUnless(oauth2_provider, 'django-oauth2-provider not installed')
     def test_post_form_with_expired_access_token_failing_auth(self):
-        """Ensure POSTing with expired access token fails with an 'Invalid token' error"""
+        """Ensure POSTing with expired access token fails with a 'token_expired' error"""
         self.access_token.expires = datetime.datetime.now() - datetime.timedelta(seconds=10)  # 10 seconds late
         self.access_token.save()
         auth = self._create_authorization_header()
         response = self.csrf_client.post('/oauth2-test/', HTTP_AUTHORIZATION=auth)
-        self.assertIn(response.status_code, (status.HTTP_401_UNAUTHORIZED, status.HTTP_403_FORBIDDEN))
-        self.assertIn('Invalid token', response.content)
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+        self.assertIn(authentication.OAUTH2_TOKEN_ERROR_EXPIRED, response.content)
+
+    @unittest.skipUnless(oauth2_provider, 'django-oauth2-provider not installed')
+    def test_post_for_with_nonexistent_access_token_failing_auth(self):
+        """Ensure POSTing with a nonexistent access token fails with a 'token_nonexistent' error"""
+        auth = self._create_authorization_header(token='notatoken')
+        response = self.csrf_client.post('/oauth2-test/', HTTP_AUTHORIZATION=auth)
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+        self.assertIn(authentication.OAUTH2_TOKEN_ERROR_NONEXISTENT, response.content)
+
+    @unittest.skipUnless(oauth2_provider, 'django-oauth2-provider not installed')
+    def test_post_for_with_malformed_access_token_failing_auth(self):
+        """Ensure POSTing with an invalid access token fails with a 'token_malformed' error"""
+        auth = self._create_authorization_header(token='invalid token')
+        response = self.csrf_client.post('/oauth2-test/', HTTP_AUTHORIZATION=auth)
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+        self.assertIn(authentication.OAUTH2_TOKEN_ERROR_MALFORMED, response.content)
+
+    @unittest.skipUnless(oauth2_provider, 'django-oauth2-provider not installed')
+    def test_post_for_with_missing_access_token_failing_auth(self):
+        """Ensure POSTing with an invalid access token fails with a 'token_malformed' error"""
+        auth = self._create_authorization_header(token='')
+        response = self.csrf_client.post('/oauth2-test/', HTTP_AUTHORIZATION=auth)
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+        self.assertIn(authentication.OAUTH2_TOKEN_ERROR_NOT_PROVIDED, response.content)
```

Comment:
To support DRY, either:
A. create an internal test helper method that executes these 4 lines for each of the above tests, and have each test use the helper method.  Or
B. use DDT and pass in the token value and expected error_code.